stumpy/stump.py

```diff
@@ -399,47 +460,64 @@ def _stump(
             diags_ranges[thread_idx, 1],
             thread_idx,
             ρ,
+            ρL,
+            ρR,
             I,
+            IL,
+            IR,
             ignore_trivial,
+            k,
         )
 
     # Reduction of results from all threads
     for thread_idx in range(1, n_threads):
-        for i in prange(l):
-            if ρ[0, i, 0] < ρ[thread_idx, i, 0]:
-                ρ[0, i, 0] = ρ[thread_idx, i, 0]
-                I[0, i, 0] = I[thread_idx, i, 0]
-            # left pearson correlation and left matrix profile indices
-            if ρ[0, i, 1] < ρ[thread_idx, i, 1]:
-                ρ[0, i, 1] = ρ[thread_idx, i, 1]
-                I[0, i, 1] = I[thread_idx, i, 1]
-            # right pearson correlation and right matrix profile indices
-            if ρ[0, i, 2] < ρ[thread_idx, i, 2]:
-                ρ[0, i, 2] = ρ[thread_idx, i, 2]
-                I[0, i, 2] = I[thread_idx, i, 2]
-
-    # Convert pearson correlations to distances
-    p_norm = np.abs(2 * m * (1 - ρ[0, :, :]))
+        # update top-k arrays
+        core._merge_topk_ρI(ρ[0], ρ[thread_idx], I[0], I[thread_idx])
+
+        # update left matrix profile and  matrix profile indices
+        mask = ρL[0] < ρL[thread_idx]
+        ρL[0][mask] = ρL[thread_idx][mask]
```

Comment:
we need to do chain slicing. Otherwise, `numba` gives error because of advance indexing. So, I assume this is still better than `np.where(...)`.